datastore/src/main/java/com/ds/datastore/BookController.java

```diff
@@ -88,6 +110,30 @@ public BookController(BookRepository repository, BookModelAssembler assembler, B
         return CollectionModel.of(entityModelList, linkTo(methodOn(BookController.class)).withSelfRel());
     }
 
+    private CollectionModel<EntityModel<Book>> multipleToLeader(BookArray array) throws Exception {
+        String address = this.map.get(leader.getLeader());
+        address = address.substring(0,address.lastIndexOf("/") + 1) + "books";
```

Comment:
since we are using this twice, I think it would be really nice to refactor into a method